src/dune_lang/pform.ml

```diff
@@ -628,6 +634,7 @@ module Env = struct
          ; "ocaml-config", macro Ocaml_config
          ; "env", since ~version:(1, 4) Macro.Env
          ; "coq", macro Coq_config
+         ; "oxcaml", since ~version:(3, 20) Macro.Oxcaml
```

Comment:
Shouldn't you be checking this against the oxcaml extension? I think you will need to extend `Since` to be more generic to support that.

Reply:
The problem I encountered is: to use `extensions`, I need to deal with the `Dune_project` module because this is where the information about extensions are stored. However `Dune_project` already depends on `pform` so I can't do anything there without creating a cycle. The idea was to move the check to where all of this would be instantiated, more particularly when we are expanding the values. At this point of the code, we know everything is instantiated. 

You made me realized it doesn't need to be a macro as I can also get the `Dune_project.t` from the `Var` expander.

Reply:
@rgrinberg I have moved it back to the variable and port the check directly inside the expander. Would it be better this way?